vnext/local-cli/generator-windows/templates/cs/proj/MyApp.csproj

```diff
@@ -154,25 +153,44 @@
     </PackageReference>
   </ItemGroup>
   <ItemGroup>
-    <ProjectReference Include="$(ReactNativeWindowsDir)\Microsoft.ReactNative\Microsoft.ReactNative.vcxproj">
-      <Project>{f7d32bd0-2749-483e-9a0d-1635ef7e3136}</Project>
-      <Name>Microsoft.ReactNative</Name>
-    </ProjectReference>
-    <PackageReference Include="Microsoft.UI.Xaml">
-      <Version>2.3.191129002</Version>
-    </PackageReference>
+    <Reference Include="Microsoft.UI.Xaml, Version=255.255.255.255, Culture=neutral, PublicKeyToken=null, ContentType=WindowsRuntime">
```

Comment:
Maybe not in scope for this PR, but I'm still wondering if it's possible to move packages internal to the project. It would require some craftiness, but I think it should be possible. 

I.e. we have some project and packages.config internally, declare a dependency on that project externally and add it to the template solution, than have the template project copy artifacts from the internal project.

It lets us pull a bit more out of the template.

Reply:
If we use [PacakgeReference](https://docs.microsoft.com/en-us/nuget/consume-packages/migrate-packages-config-to-package-reference), this becomes a lot cleaner:
```xml
<PackageReference Include="Microsoft.UI.Xaml" version="255.255.255.255"/>
```

In fact if we create our nuget package and use 
```xml
<PackageReference Include="Microsoft.ReactNative.Uwp.Managed" version="0.63-preview.1"/>
```
and declare dependencies on Microsoft.UI.Xaml and Microsoft.UI.Xaml.Design (as a dev-only dependency) Then users will not even have to declare this as the msbuild/nuget model is to handle all references transitively for compiliatoin instead of the expected deployment only.



---
In reply to: [421389921](https://github.com/microsoft/react-native-windows/pull/4816#discussion_r421389921) [](ancestors = 421389921)

Reply:
of course, the template can then ommit these dependencies, but this won't prevent users from adding then through the nuget UI to get specific versions they want. Without these when they upgrade, it will just fine and when we bump the ms.ui.xaml version they will get it automatically. If they used the odl one they will likely get a nuget error (or at least a warning) about inconsistent versions and that they are downgrading. My assumption is that ms.ui.xaml is going to provide a high level of compatibility, so upgrading users who don't want a specific version automatically is advantageous else we will always have ot worry about back compate if we want to use a new XAML version. Then we break everybody! if we remove these (eventually when we get to a package) then we only break people who explicitly chose a differetn versoin manually.

Reply:
MY goal was to normalize around using packages.config from a upgrade maintainability point of view - unfortunately, for C# projects, Microsoft.UI.Xaml inserts an unnecessary reference to Microsoft.UI.XamlkDesign, which is x86 only, and therefore breaks builds on other flavors. So I've switched back to PackageReferences

Reply:
Also, ms.ui.xaml is going to have major changes with the release of WinUI 3.

Reply:
@NickGerleman I do not want a hidden internal project and copying artifacts. it's unnecessary and weird